From 278cddf4bffdc31d1e0b13815f9eb468c2abc0f3 Mon Sep 17 00:00:00 2001
From: Nayak, Rajendra <rnayak@ti.com>
Date: Fri, 13 Feb 2009 11:30:57 +0530
Subject: [PATCH 353/426] OMAP3: PM: Update voltage levels for OPP1/2 on VDD1/2

This patch updates the voltage levels for VDD1 OPP1/2 and
VDD2 OPP1/2 according to the latest operating condition
addendum for 3430.

The new voltage levels at various OPP's for VDD1/2 are as below

VDD1 OPP1 0.975v
VDD1 OPP2 1.050v
VDD1 OPP3 1.200v
VDD1 OPP4 1.270v
VDD1 OPP5 1.350v

VDD2 OPP1 0.975v
VDD2 OPP2 1.050v
VDD2 OPP3 1.150v

Formula used to calulate the hex values
Value(in decimals)*12.5 + 600 mV.

The patch applies on the latest pm head and is validated on 3430SDP.

Signed-off-by: Rajendra Nayak <rnayak@ti.com>
(cherry picked from commit dce2c221dfaf4de0cae85ebe4f815a62047a06cf)
(cherry picked from commit b1e975bc52ef9344d86cbed2ffa2bf70e80585da)
---
 arch/arm/mach-omap2/omap3-opp.h |   12 ++++++------
 arch/arm/mach-omap2/pm34xx.c    |   16 ++++++++--------
 2 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/arch/arm/mach-omap2/omap3-opp.h b/arch/arm/mach-omap2/omap3-opp.h
index 5540e5b..4bb165e 100644
--- a/arch/arm/mach-omap2/omap3-opp.h
+++ b/arch/arm/mach-omap2/omap3-opp.h
@@ -24,9 +24,9 @@
 static struct omap_opp omap3_mpu_rate_table[] = {
 	{0, 0, 0},
 	/*OPP1*/
-	{S125M, VDD1_OPP1, 0x18},
+	{S125M, VDD1_OPP1, 0x1E},
 	/*OPP2*/
-	{S250M, VDD1_OPP2, 0x20},
+	{S250M, VDD1_OPP2, 0x24},
 	/*OPP3*/
 	{S500M, VDD1_OPP3, 0x30},
 	/*OPP4*/
@@ -38,9 +38,9 @@ static struct omap_opp omap3_mpu_rate_table[] = {
 static struct omap_opp omap3_l3_rate_table[] = {
 	{0, 0, 0},
 	/*OPP1*/
-	{0, VDD2_OPP1, 0x18},
+	{0, VDD2_OPP1, 0x1E},
 	/*OPP2*/
-	{S83M, VDD2_OPP2, 0x20},
+	{S83M, VDD2_OPP2, 0x24},
 	/*OPP3*/
 	{S166M, VDD2_OPP3, 0x2C},
 };
@@ -48,9 +48,9 @@ static struct omap_opp omap3_l3_rate_table[] = {
 static struct omap_opp omap3_dsp_rate_table[] = {
 	{0, 0, 0},
 	/*OPP1*/
-	{S90M, VDD1_OPP1, 0x18},
+	{S90M, VDD1_OPP1, 0x1E},
 	/*OPP2*/
-	{S180M, VDD1_OPP2, 0x20},
+	{S180M, VDD1_OPP2, 0x24},
 	/*OPP3*/
 	{S360M, VDD1_OPP3, 0x30},
 	/*OPP4*/
diff --git a/arch/arm/mach-omap2/pm34xx.c b/arch/arm/mach-omap2/pm34xx.c
index 70bc793..7c00044 100644
--- a/arch/arm/mach-omap2/pm34xx.c
+++ b/arch/arm/mach-omap2/pm34xx.c
@@ -1054,16 +1054,16 @@ err2:
 }
 
 /* PRM_VC_CMD_VAL_0 specific bits */
-#define OMAP3430_VC_CMD_VAL0_ON				(0x3 << 4)
-#define OMAP3430_VC_CMD_VAL0_ONLP			(0x3 << 3)
-#define OMAP3430_VC_CMD_VAL0_RET			(0x3 << 3)
-#define OMAP3430_VC_CMD_VAL0_OFF			(0x3 << 4)
+#define OMAP3430_VC_CMD_VAL0_ON		0x30
+#define OMAP3430_VC_CMD_VAL0_ONLP	0x1E
+#define OMAP3430_VC_CMD_VAL0_RET	0x1E
+#define OMAP3430_VC_CMD_VAL0_OFF	0x30
 
 /* PRM_VC_CMD_VAL_1 specific bits */
-#define OMAP3430_VC_CMD_VAL1_ON				(0xB << 2)
-#define OMAP3430_VC_CMD_VAL1_ONLP			(0x3 << 3)
-#define OMAP3430_VC_CMD_VAL1_RET			(0x3 << 3)
-#define OMAP3430_VC_CMD_VAL1_OFF			(0xB << 2)
+#define OMAP3430_VC_CMD_VAL1_ON		0x2C
+#define OMAP3430_VC_CMD_VAL1_ONLP	0x1E
+#define OMAP3430_VC_CMD_VAL1_RET	0x1E
+#define OMAP3430_VC_CMD_VAL1_OFF	0x2C
 
 static void __init configure_vc(void)
 {
-- 
1.6.2.4

